desktop/libs/notebook/src/notebook/connectors/trino_tests.py

```diff
@@ -0,0 +1,239 @@
+      # Mock TrinoQuery object and its methods
+      mock_query_instance = TrinoQuery.return_value
+      mock_query_instance.query = "SELECT * FROM test_table"
+      mock_query_instance.execute.return_value = MagicMock(next_uri=None, id='123', rows=[], columns=[])
+
+
+      mock_trino_request = MagicMock()
+      self.trino_api.trino_request = mock_trino_request
+
+      # Configure the MagicMock object to return expected responses
+      mock_trino_request.get.return_value = MagicMock()
+      mock_trino_request.process.return_value = MagicMock(stats={'state': 'FINISHED'}, next_uri='http://url', id=123, rows=[])
+
+      # Call the execute method
+      result = self.trino_api.execute(notebook={}, snippet={'database': 'test_db', 'statement': 'SELECT * FROM test_table;'})
+
+      self.assertEqual(result['row_count'], 0)
+      self.assertEqual(result['next_uri'], 'http://url')
+      self.assertIsNone(result['sync'])
+      self.assertTrue(result['has_result_set'])
+      self.assertEqual(result['guid'], 123)
+      self.assertEqual(result['result']['has_more'], True)
+      self.assertEqual(result['result']['data'], [])
+      self.assertEqual(result['result']['meta'], [])
+      self.assertEqual(result['result']['type'], 'table')
+
+
+  def test_fetch_result(self):
+    # Mock TrinoRequest object and its methods
+    mock_trino_request = MagicMock()
+    self.trino_api.trino_request = mock_trino_request
+
+    # Configure the MagicMock object to return expected responses
+    mock_trino_request.get.return_value = MagicMock()
+    _columns = [{'comment': '', 'name': 'test_column1', 'type': 'str'}, {'comment': '', 'name': 'test_column2', 'type': 'str'}]
+
+    mock_trino_request.process.side_effect = [
+      MagicMock(
+        stats={'state': 'FINISHED'}, next_uri='http://url', id=123,
+        rows=[['value1', 'value2'], ['value3', 'value4']], columns=_columns
+      ),
+      MagicMock(
+        stats={'state': 'FINISHED'}, next_uri='http://url1', id=124,
+        rows=[['value5', 'value6'], ['value7', 'value8']], columns=_columns
+      ),
+      MagicMock(
+        stats={'state': 'FINISHED'}, next_uri=None, id=125,
+        rows=[['value9', 'value10'], ['value11', 'value12']], columns=_columns
+      )
+    ]
+
+    # Call the fetch_result method
+    result = self.trino_api.fetch_result(
+      notebook={}, snippet={'result': {'handle': {'next_uri': 'http://url', 'result': {'data': []}}}}, rows=0, start_over=False
+    )
+
+    self.assertEqual(result['row_count'], 94)
+    self.assertIsNone(result['next_uri'])
+    self.assertFalse(result['has_more'])
+    self.assertEqual(len(result['data']), 6)
+    self.assertEqual(result['data'][0], ['value1', 'value2'])
+    self.assertEqual(len(result['meta']), 2)
+    self.assertEqual(result['meta'][0]['name'], 'test_column1')
+    self.assertEqual(result['meta'][0]['type'], 'str')
+    self.assertEqual(result['meta'][0]['comment'], '')
```

Comment:
Same check the whole result dict

Reply:
I wanted to check some other things as well, but I've updated them now.